### ISSUE TITLE
Intervention tests (test_intervention.py) fail after unrelated changes.

### ISSUE THREAD

**Issue by ghart-IDM** [InstituteforDiseaseModeling/covasim#204]
**Describe the bug**

In the current develop branch (head: 0c73031998091168aeb3ceff740cdbff5004a3aa) or with the code in pull request #434 the test_change_beta_layers_clustered test and number of the test sensitivity tests fail as the random seed, viral load distribution, and/or beta distribution change and potentially other parameters that have no effect on the things being tested. I have some thoughts on how to make these tests more robust.

**To reproduce**

Easiest way to see this is take the current head (0c73031998091168aeb3ceff740cdbff5004a3aa) and change rand_seed in parameters.py to 2 instead of one and then run test_interventions.py:

```bash
#change rand_seed in parameters.py to 2
pytest test_interventions.py
```

**Expected behavior**

Pass all tests

**Screenshots or outputs**
```
(base) C:\Users\ghart\OneDrive - IDMOD\Codes\covasim\tests\unittests>pytest test_interventions.py
================================================= test session starts =================================================
platform win32 -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: C:\Users\ghart\OneDrive - IDMOD\Codes\covasim
plugins: arraydiff-0.3, doctestplus-0.3.0, openfiles-0.3.2, remotedata-0.3.2, celery-4.4.2
collected 15 items

test_interventions.py ...F..s.s....F.                                                                            [100%]

====================================================== FAILURES =======================================================
_________________________________ InterventionTests.test_change_beta_layers_clustered _________________________________

self = <test_interventions.InterventionTests testMethod=test_change_beta_layers_clustered>

    def test_change_beta_layers_clustered(self):
        self.is_debugging = False
        initial_infected = 10
        params = {
            SimKeys.number_agents: 5000,
            SimKeys.number_simulated_days: 60,
            SimKeys.initial_infected_count: initial_infected
        }
        self.set_simulation_parameters(params_dict=params)
        day_of_change = 25
        change_multipliers = [0.0]
        layer_keys = ['c','h','s','w']

        sequence_days = []
        sequence_interventions = []

        for k in layer_keys: # Zero out one layer at a time
            day_of_change += 5
            self.intervention_set_changebeta(
                days_array=[day_of_change],
                multiplier_array=change_multipliers,
                layers=[k]
            )
            sequence_days.append(day_of_change)
            sequence_interventions.append(self.interventions)
            self.interventions = None
            pass
        self.intervention_build_sequence(day_list=sequence_days,
                                         intervention_list=sequence_interventions)
        self.run_sim(population_type='clustered')
        last_intervention_day = sequence_days[-1]
        first_intervention_day = sequence_days[0]
        cum_infections_channel= self.get_full_result_channel(ResultsKeys.infections_cumulative)
        self.assertGreater(cum_infections_channel[sequence_days[0]-1],
                           initial_infected,
                           msg=f"Before intervention at day {sequence_days[0]}, there should be infections happening.")
        self.assertGreater(cum_infections_channel[last_intervention_day],
                           cum_infections_channel[first_intervention_day],
                           msg=f"Cumulative infections should grow with only some layers enabled.")
        self.assertEqual(cum_infections_channel[last_intervention_day],
                         cum_infections_channel[-1],
>                        msg=f"With all layers at 0 beta, the cumulative infections at {last_intervention_day}"
                             f" should be the same as at the end.")
E       AssertionError: 38.0 != 44.0 : With all layers at 0 beta, the cumulative infections at 45 should be the same as at the end.

test_interventions.py:174: AssertionError
------------------------------------------------ Captured stdout call -------------------------------------------------
Initializing sim with 5000 people for 60 days
Could not sanitize "<covasim.interventions.change_beta object at 0x0000027A09AAF898>" <class 'covasim.interventions.change_beta'> (Object of type change_beta is not JSON serializable), converting to string instead
Could not sanitize "<covasim.interventions.change_beta object at 0x0000027A09AAF940>" <class 'covasim.interventions.change_beta'> (Object of type change_beta is not JSON serializable), converting to string instead
Could not sanitize "<covasim.interventions.change_beta object at 0x0000027A09AAF9B0>" <class 'covasim.interventions.change_beta'> (Object of type change_beta is not JSON serializable), converting to string instead
Could not sanitize "<covasim.interventions.change_beta object at 0x0000027A09AAFA20>" <class 'covasim.interventions.change_beta'> (Object of type change_beta is not JSON serializable), converting to string instead
____________________________________ InterventionTests.test_test_prob_sensitivity _____________________________________

self = <test_interventions.InterventionTests testMethod=test_test_prob_sensitivity>, subtract_today_recoveries = True

    def test_test_prob_sensitivity(self, subtract_today_recoveries=True):
        self.is_debugging = False
        params = {
            SimKeys.number_agents: 5000,
            SimKeys.number_simulated_days: 31
        }
        self.set_simulation_parameters(params_dict=params)

        symptomatic_probability_of_test = 1.0
        test_sensitivities = [0.9, 0.7, 0.6, 0.2]
        test_delay = 0
        start_day = 30

        for sensitivity in test_sensitivities:
            self.intervention_set_test_prob(symptomatic_prob=symptomatic_probability_of_test,
                                            test_sensitivity=sensitivity,
                                            test_delay=test_delay,
                                            start_day=start_day)
            self.run_sim()
            first_day_diagnoses = self.get_full_result_channel(
                channel=ResultsKeys.diagnoses_at_timestep
            )[start_day]
            target_count = self.get_full_result_channel(
                channel=ResultsKeys.symptomatic_at_timestep
            )[start_day]
            if subtract_today_recoveries:
                recoveries_today = self.get_full_result_channel(
                    channel=ResultsKeys.recovered_at_timestep
                )[start_day]
                target_count = target_count - recoveries_today
            ideal_diagnoses = target_count * sensitivity

            standard_deviation = sqrt(sensitivity * (1 - sensitivity) * target_count)
            # 95% confidence interval
            min_tolerable_diagnoses = ideal_diagnoses - 2 * standard_deviation
            max_tolerable_diagnoses = ideal_diagnoses + 2 * standard_deviation

            if self.is_debugging:
                print(f"\tMax: {max_tolerable_diagnoses} \n"
                      f"\tMin: {min_tolerable_diagnoses} \n"
                      f"\tTarget: {target_count} \n"
                      f"\tPrevious day Target: {self.get_full_result_channel(channel=ResultsKeys.symptomatic_at_timestep)[start_day -1 ]} \n"
                      f"\tSensitivity: {sensitivity} \n"
                      f"\tIdeal: {ideal_diagnoses} \n"
                      f"\tActual diagnoses: {first_day_diagnoses}\n")
            self.assertGreaterEqual(first_day_diagnoses, min_tolerable_diagnoses,
                                    msg=f"Expected at least {min_tolerable_diagnoses} diagnoses with {target_count}"
                                        f" symptomatic and {sensitivity} sensitivity. Got {first_day_diagnoses}"
                                        f" diagnoses, which is too low.")
            self.assertLessEqual(first_day_diagnoses, max_tolerable_diagnoses,
>                                msg=f"Expected no more than {max_tolerable_diagnoses} diagnoses with {target_count}"
                                     f" symptomatic and {sensitivity} sensitivity. Got {first_day_diagnoses}"
                                     f" diagnoses, which is too high.")
E           AssertionError: 34.0 not less than or equal to 33.47242920512754 : Expected no more than 33.47242920512754 diagnoses with 123.0 symptomatic and 0.2 sensitivity. Got 34.0 diagnoses, which is too high.

test_interventions.py:534: AssertionError
------------------------------------------------ Captured stdout call -------------------------------------------------
Initializing sim with 5000 people for 31 days
Initializing sim with 5000 people for 31 days
Initializing sim with 5000 people for 31 days
Initializing sim with 5000 people for 31 days
================================================== warnings summary ===================================================
C:\ProgramData\Anaconda3\lib\site-packages\parso\python\tree.py:46
  C:\ProgramData\Anaconda3\lib\site-packages\parso\python\tree.py:46: DeprecationWarning:

  Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working

C:\ProgramData\Anaconda3\lib\site-packages\ipywidgets\widgets\widget.py:280
  C:\ProgramData\Anaconda3\lib\site-packages\ipywidgets\widgets\widget.py:280: DeprecationWarning:

  Widget registration using a string name has been deprecated. Widget registration now uses a plain `@register` decorator.

C:\ProgramData\Anaconda3\lib\site-packages\ipywidgets\widgets\widget_upload.py:45
  C:\ProgramData\Anaconda3\lib\site-packages\ipywidgets\widgets\widget_upload.py:45: DeprecationWarning:

  Traits should be given as instances, not types (for example, `Int()`, not `Int`). Passing types is deprecated in traitlets 4.1.

C:\ProgramData\Anaconda3\lib\site-packages\ipywidgets\widgets\widget_upload.py:46
  C:\ProgramData\Anaconda3\lib\site-packages\ipywidgets\widgets\widget_upload.py:46: DeprecationWarning:

  Traits should be given as instances, not types (for example, `Int()`, not `Int`). Passing types is deprecated in traitlets 4.1.

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================= 2 failed, 11 passed, 2 skipped, 4 warnings in 15.26 seconds =============================
```

**Platform (please complete the following information):**
 - Covasim version: head 0c73031998091168aeb3ceff740cdbff5004a3aa or pull request  #434 
 - OS: Linux & Windows
 - Other platform information: using Anaconda Python

**Comment by CWiswell-IDM** [comment]
Working on it. 
In the first case: Apparently I wasn't using sequences properly and got lucky on a particular random seed.  Converting these to lists of interventions rather than sequences, and should consider adding a specific test of the sequence intervention when I understand it better.

In the second case, it looks like the number of diagnoses was a hair out of 2 standard deviations.  Thinking about it...

**Comment by cliffckerr** [comment]
Thanks @CWiswell-IDM ! I just pushed an update to master that will affect this. There will be another big change to parameter values (but not names or structure) coming in the next couple hours, so results will change again.

**Comment by CWiswell-IDM** [comment]
I think that there is a real defect here somewhere, as we seem to consistently be diagnosing more people than we should.  Sweeping the test_sensitivity scenario, I consistently get some results that seem too high, and never get results that would seem too low.  I'll try it without removing the recoveries...

**Comment by CWiswell-IDM** [comment]
Okay. If I do not remove recoveries from the ideal_diagnoses, we are doing much better. If I allow one diagnoses more or less in tolerance (to account for rounding errors) we seem to be doing fine...  PR incoming.

**Comment by cliffckerr** [comment]
Closed by #207
